app/main.py

```diff
@@ -1,2 +1,54 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    def __init__(
+        self, initial_capacity: int = 8, load_factor: float = 2 / 3
+    ) -> None:
+        self.initial_capacity = initial_capacity
+        self.load_factor = load_factor
+        self.capacity = initial_capacity
+        self.size = 0
+        self.buckets = [[] for _ in range(initial_capacity)]
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        bucket_index = self._get_bucket_index(key)
+        bucket = self.buckets[bucket_index]
+
+        for i, (k, _) in enumerate(bucket):
+            if k == key:
+                bucket[i] = (key, value)
+                return
```

Comment:
Your code isn't handling collissions properly
You should regenerate the index when the collission occurs

Reply:
Also we need to store hash in buckets